### PR TITLE
Add load/save map feature

### PR DESCRIPTION
**Pull request by hello-atharva**
## Description

Adds functionality to save/load maps in ORB_SLAM3.  Requires a rebuild of [ORB_SLAM3](https://github.com/hello-atharva/ORB_SLAM3) and its [python wrapper](https://github.com/hello-atharva/ORB_SLAM3-PythonBindings). It internally calls `SaveAtlasToFile` method of ORB_SLAM's C++ backend.

To run the demo app, execute

```
python3 -m stretch.app.map_saver --robot_ip $ROBOT_IP
```

## Checklist

- \[ \] I have performed a self-review of my code
- \[ \] If it is a core feature, I have added thorough tests
- \[ \] I have added documentation for the changes
- \[ \] I have updated the README file if necessary
- \[ \] I have run on hardware if necessary

## Screenshots (if applicable)

Add any relevant screenshots or screen recordings to help reviewers understand the changes.

## Additional context

Add any other context or information about the pull request here.

### CODE REVIEW

File: src/stretch/agent/zmq_client.py [hunk 1]
```diff
@@ -726,6 +726,14 @@ def at_goal(self) -> bool:
                 return False
             return self._state["at_goal"]
 
+    def save_map(self, filename: str):
```
Comment (hello-cpaxton) [inline src/stretch/agent/zmq_client.py:729]: add docstring?

File: src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/orbslam3.py [hunk 2]
```diff
@@ -85,6 +86,12 @@ def __init__(self):
             CameraInfo, "/camera/color/camera_info", self.camera_info_callback, 1
         )
 
+        # Create a service to save the map
+        self.save_map_service = self.create_service(SaveMap, "/save_map", self.save_map_callback)
```
Comment (hello-cpaxton) [inline src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/orbslam3.py:90]: cool

File: src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/orbslam3.py [hunk 3]
```diff
@@ -144,6 +151,24 @@ def accel_callback(self, msg):
     def gyro_callback(self, msg):
         self.gyro_data = msg
 
+    def save_map_callback(self, request, response):
+        if self.slam is not None:
+            map_file = request.map_url
```
Comment (hello-cpaxton) [inline src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/orbslam3.py:156]: this should also work for hector right?

File: src/stretch_ros2_bridge/stretch_ros2_bridge/remote/api.py [hunk 4]
```diff
@@ -230,6 +232,12 @@ def get_base_pose(self) -> np.ndarray:
         """Get the robot's base pose as XYT."""
         return self.nav.get_base_pose()
 
+    def load_map(self, filename: str):
```
Comment (hello-cpaxton) [inline src/stretch_ros2_bridge/stretch_ros2_bridge/remote/api.py:235]: data

Comment (hello-cpaxton) [reply src/stretch_ros2_bridge/stretch_ros2_bridge/remote/api.py:235]: docstring i mean

File: src/stretch_ros2_bridge/stretch_ros2_bridge/remote/modules/mapping.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from nav2_msgs.srv import LoadMap, SaveMap
+
+from .abstract import AbstractControlModule
+
+
+class StretchMappingClient(AbstractControlModule):
```
Comment (hello-cpaxton) [inline src/stretch_ros2_bridge/stretch_ros2_bridge/remote/modules/mapping.py:20]: how does this relate to the map saver https://github.com/hello-robot/stretch_ai/blob/main/src/stretch_ros2_bridge/stretch_ros2_bridge/ros/map_saver.py

**Comment by hello-atharva** [comment]
CI is passing

**Review by hello-cpaxton — APPROVED** [review]
LGTM

**Pull request merged**